src/react/components/pages/modelCompose/modelCompose.tsx

```diff
@@ -0,0 +1,305 @@
+                            />
+                    </div>
+                    {this.state.isComposing ?
+                    <Spinner
+                        label="Model is composing, please wait..."
+                        className="compose-spinner"
+                        theme={getDefaultDarkTheme()}
+                        size={SpinnerSize.large}>
+                    </Spinner> :
+                    <DetailsList
+                        items = {modelList}
+                        compact={isCompactMode}
+                        columns={columns}
+                        selectionMode={SelectionMode.multiple}
+                        layoutMode={DetailsListLayoutMode.justified}
+                        isHeaderVisible={true}
+                        selection={this.selection}
+                        selectionPreservedOnEmptyClick={true}>
+                    </DetailsList>
+                    }
+                </Customizer>
+            </Fabric>
+        );
+    }
+
+    private getModelList = async () => {
+        try {
+            const res = await this.getReponse();
+            const modelList = res.data.modelList;
+            this.setState({
+                modelList,
+            });
+        } catch (error) {
+            console.log(error);
+        }
+    }
+
+    private async getReponse() {
+        const baseURL = url.resolve(
+            this.props.project.apiUriBase,
+            constants.apiModelsPath,
+        );
+
+        try {
+            return await ServiceHelper.getWithAutoRetry(
+                baseURL,
+                {},
+                this.props.project.apiKey as string,
+            );
+        } catch (err) {
+            ServiceHelper.handleServiceError(err);
+        }
+    }
+
+    private handleColumnClick = (event: React.MouseEvent<HTMLElement>, column: IColumn): void => {
+        const {columns, modelList} = this.state;
+        const newColumns: IColumn[] = columns.slice();
+        const currColumn: IColumn = newColumns.filter((col) => column.key === col.key)[0];
+        newColumns.forEach((newCol: IColumn) => {
+            if (newCol === currColumn) {
+                currColumn.isSortedDescending = !currColumn.isSortedDescending;
+                currColumn.isSorted = true;
+            } else {
+                newCol.isSorted = false;
+                newCol.isSortedDescending = true;
+            }
+        });
+        const newList = this.copyAndSort(modelList, currColumn.fieldName!, currColumn.isSortedDescending);
+        this.setState({
+            columns: newColumns,
+            modelList: newList,
+        });
+    }
+
+    private copyAndSort(modelList: IModel[], columnKey: string, isSortedDescending?: boolean): IModel[] {
+        const key = columnKey;
+        if (key === "createdatetime" || key === "lastupdateddatetime") {
+            return (modelList.slice(0)
+            .sort((a, b): number => {
+                if (isSortedDescending) {
```

Comment:
compareFunction must be consistant for the same two values, means a == b, must return 0, otherwise both compare(a, b) and compare(b, a) will returns -1 which will cause potential bug.
Try this:
const cmp = (new Date(b.createdDateTime)).getTime() -  (new Date(a.createdDateTime)).getTime();
return isSortedDescending ? cmp : -cmp;